### PR TITLE
Migrate from setup.cfg to pyproject.toml completely

### PR DESCRIPTION
**Pull request by alex**
One less file?

### CODE REVIEW

**Comment by alex** [comment]
minimum setuptools version this requires is py37 only, so alas, this is on hold

**Comment by alex** [comment]
with https://github.com/pyca/infra/pull/449 this will be green (though not mergable until 41)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ jobs:
           - {VERSION: "3.11", TOXENV: "docs", OPENSSL: {TYPE: "openssl", VERSION: "3.0.7"}}
           - {VERSION: "pypy-3.8", TOXENV: "pypy3-nocoverage"}
           - {VERSION: "pypy-3.9", TOXENV: "pypy3-nocoverage"}
+          - {VERSION: "3.11", TOXENV: "py311", OPENSSL: {TYPE: "openssl", VERSION: "1.1.1c"}}
```
Comment (reaperhulk) [inline .github/workflows/ci.yml:34]: Should we be adding this job now so we can drop focal without coverage loss later? 😄

File: pyproject.toml [hunk 2]
```diff
@@ -1,15 +1,97 @@
...
+[tool.setuptools.packages.find]
+where = ["src"]
+exclude = [
+    "_cffi_src",
+    "_cffi_src.*",
+]
+
+[tool.setuptools.dynamic]
+version = {attr = "cryptography.__version__"}
+readme = {file = "README.rst", content-type = "text/x-rst"}
```
Comment (Viicos) [inline pyproject.toml:70]: iirc, `readme` doesn't have to be dynamic if an extension (`.rst` in this case) is provided. See https://packaging.python.org/en/latest/specifications/declaring-project-metadata/#readme for reference

**Review by reaperhulk — APPROVED** [review]

**Pull request merged**